src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml

```diff
@@ -41,6 +41,7 @@ well as menu structures (for main menu and popup menus).
             <cmd refid="newSourceDoc"/>
             <separator/>
             <cmd refid="newRMarkdownDoc"/>
+            <cmd refid="newRShinyApp"/>
```

Comment:
You'll also want to add the command to the new button the global toolbar here: https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/application/ui/GlobalToolbar.java#L50-L62